### DISCUSSION TITLE
mmap memory issues - solution

### DISCUSSION THREAD

**Discussion by treeder** [gochain-io/gochain#443] [General]
If you get an mmap `cannot allocate memory` error and your node fails to sync, please try this:

* `sudo sysctl -w vm.max_map_count=262144`
* `sudo nano /etc/sysctl.conf` add `vm.max_map_count = 262144` 
* `sudo systemctl restart docker`

**Comment by treeder** [comment]
In addition, ensure the `--cache` flag in `docker-compose.yml` is set to ~16000 (half total memory assuming a 32GB memory instance).

**Reply by zelonH** [reply]
We deployed directly with docker, not through docker-compose, what should we do

**Reply by treeder** [reply]
It's for the `gochain` program: https://github.com/gochain/docs/blob/master/public-network/signers/nodes/mainnet/docker-compose.yml#L11

So however you are running the `gochain` binary, add the `--cache flag`

**Reply by zelonH** [reply]
How to add cache through docker command      , I used docker run -d --name go -v /data/go:/work/bin/go/data -p 8821:8819 go supervisord -n -c /etc/supervisord.conf to run this cache command. Add it

**Reply by zelonH** [reply]
Because we don’t have the GOCHAIN_ACCT=0x123456 configuration, we can’t use the docker-compose method

**Reply by zelonH** [reply]
We use the dockerfile command
FROM abc_base
#FROM centos

#------------------install go------------------------
 
# build go
RUN mkdir -p /work/install
RUN yum install gcc-c++ screen git wget go  make -y

RUN cd /work && wget https://dl.google.com/go/go1.15.linux-amd64.tar.gz
RUN cd /work \
  && tar zxvf go1.15.linux-amd64.tar.gz

ENV GOROOT=/work/go
ENV GOPATH=/work/install
ENV PATH=$GOROOT/bin:$GOPATH/bin:$PATH

RUN cd /work/install && git clone https://github.com/gochain-io/gochain && cd $GOPATH/gochain && git checkout v3.4.9 \
  && cd /work/install/gochain && make gochain

RUN mkdir -p /work/bin && cp /work/install/gochain/bin/gochain /work/bin

RUN echo "successfull install go"


RUN yum install -y supervisor
ADD supervisor_go.ini /etc/supervisord.d/

**Reply by treeder** [reply]
I don't think you need the GOCHAIN_ACCT unless you're signing, but regardless, somewhere you  have to be running the `gochain` bin, maybe in your supervisor files or something.

**Reply by zelonH** [reply]
I got it, thank you
